### DISCUSSION TITLE
Using RayTracing module to trace the transient position of particles in Fluid flow

### DISCUSSION THREAD

**Discussion by sb50nemo** [idaholab/moose#20846] [Q&A Modules: General] [answered]
I am trying to trace ~ 100 - 1000 mass points inside a fluid flow. I have a `UserOject` that can update the position of each mass particle with time, however, I want to output the trajectory of those particles alongside the fluid simulation.

I have checked the Ray Tracing module which has a `RayTracingExodus` where the trajectory of the rays are saved. 
Is it possible to use the same for saving the transient position of all the particles during the fluid flow? Or could you please suggest a better way to accomplish this?

**Comment by GiudGiud** [comment, marked as answer]
Hello

The RayTracingExodus should be the rays/trajectories of these displacements. And you only want the positions ?
Am I understanding this right?

Guillaume

**Reply by sb50nemo** [reply]
Hi!

Yes, I want to store the trajectories of those particles using `RayTracingExodus`. Is it possible?

**Reply by GiudGiud** [reply]
But you are not using the RayTracing module right now?

If not then no, the RayTracingExodus is meant for the RayTracing module. The good news is that if you rework your code to use ray tracing instead of the custom UO, then you get a TON of stuff working (mesh decomposition, asynchronous communication, output, integration capabilities etc)

If you are then it should just work and if it doesnt please let us know what you see in the output.

**Reply by GiudGiud** [reply]
if you are not using RayTracing and want to persist using a UO, then you should look a building a custom vector postprocessor, saving the positions for each mass point, then write a lil python visualization tool to plot positions from a CSV file. I do not recommend that

Are you doing this for the Material Point Method btw? We have had some progress looking at this with the RayTracing module, maybe we could share that with you

**Reply by sb50nemo** [reply]
Thank you very much for your response. Our initial plan was to save the positions in CSV and use python to visualize the trajectories of each particle. However, as we have more than 1000 particles, it might not be as effective. 

We are not using MPM, For the motion of those particles, we employ a kinematic equation based on Gaussian points. Could we use Ray Tracing kernels for implementing the kinematic equation?

**Reply by GiudGiud** [reply]
How do you determine the acceleration of the particle?
What comes into play? Local fields? Other particles?
By Gaussian points you mean you need to refer to a local elemental quadrature?
